### ISSUE TITLE
[Request] NixOS pills

### ISSUE THREAD

**Issue by wizzup** [NixOS/nix-pills#120]
I just finished reading all 20 pills and love them, great work! :heart: 

It would be wonderful to have something like these for NixOS.

**Comment by jonringer** [comment]
I second this motion, I would love for there to be something to ease you into nixos modules.

@Infinisil  since he has the most knowledge on the subject (besides eelco, that I'm aware of :))

**Comment by aanderse** [comment]
There are a few RFCs which need to be accepted before this document can be created/published. The RFC which aims to move away from static ids and the RFC for nix settings are the two that come to mind. I would also suggest we as a community could benefit from a discussion on when/if a user definable `dataDir` option is justified (as this is a heated topic for some people) before this document is published as well.

I very much welcome such a document as it becomes something to cite when providing PR review.

**Comment by Infinisil** [comment]
Yeah there are some things worth discussing still, but there's also many things that could be put into "NixOS pills". I have been wanting to write something like this before, but I always thought I'd do it in the form of blog posts. I'll think about it :)
